src/centrifuge.ts

```diff
@@ -409,6 +411,10 @@ export class Centrifuge extends (EventEmitter as new () => TypedEventEmitter<Cli
     })
   }
 
+  isTransportOpen() {
```

Comment:
Got your point, removed it from public API for now, changed it to the same approach with private variables

Main reason for it was easier search for usages of methods that is shared between subscriptions and client in IDE, for example PHPStorm does not find that `_setSubscribing` is used somewhere outside of subscription

I have some ideas, how such methods can be hidden from public API, but still be directly shared between subscriptions and client, if you are ok with such direction, I can come with proposal later

Reply:
Yes, please - it's very interesting, this tight coupling exists in all Centrifugo SDKs, and I have not found a clean way to solve it keeping efficiency. I thought about using interfaces instead of direct references to instances - but does not seem too different from current situation. I suppose having more isolated transport layer could help somehow, but still sometimes Client should know about Subscription internal state. So yep - would be curious to see fresh ideas here.

Reply:
If you are ok to stay with current implementation, I will prepare another PR with proposal, make take some time

Main idea is to make new class which will expose all public api and will be some kind of proxy to current centrifuge implementation, subs will be lit bit more complicated, but as soon as we already have new instance that managed public api, it may also expose some kind of proxy subscription instance to public, when newSubscription is called, but still keep original Subscription inside client

Reply:
Yes, it's of course should be separate. I can't  promise that we will go forward with that approach though even if it makes sense for JavaScript client, it still depends on many factors - attempt to have APIs and behaviour close to each other in the ecosystem, reduce complexity if possible since maintaining SDKs is hard. It may be wrapper, ot it may also be long term thing if it makes sense and solves/improves sth. We will see